homeassistant/components/blink/__init__.py

```diff
@@ -154,26 +155,19 @@ def _async_import_options_from_data_if_missing(
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload Blink entry."""
-    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
-
-    if not unload_ok:
-        return False
-
-    hass.data[DOMAIN].pop(entry.entry_id)
-
-    if len(hass.data[DOMAIN]) != 0:
-        return True
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        hass.data[DOMAIN].pop(entry.entry_id)
 
-    hass.services.async_remove(DOMAIN, SERVICE_REFRESH)
-    hass.services.async_remove(DOMAIN, SERVICE_SAVE_VIDEO)
-    hass.services.async_remove(DOMAIN, SERVICE_SEND_PIN)
+        hass.services.async_remove(DOMAIN, SERVICE_REFRESH)
+        hass.services.async_remove(DOMAIN, SERVICE_SAVE_VIDEO)
+        hass.services.async_remove(DOMAIN, SERVICE_SEND_PIN)
```

Comment:
Why do we only remove these conditionally? I see previously we wouldn't remove these if there are still entries left

Reply:
it was conditional, just in a strange way:
https://github.com/home-assistant/core/blob/04c0bca487550d91ea8d8b8430d5a1c3b4b9ee90/homeassistant/components/blink/__init__.py#L157-L162
I refactored it to match the "normal" way this done (from the scaffold)

Reply:
Oh - now I see.  If the same domain has something left, don't remove the services.  I don't think that's possible.  How do I have 2 entries for the domain as it is?

Reply:
I suppose there could be 2 blink accounts...